bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/api/ZWaveJSClient.java

```diff
@@ -0,0 +1,399 @@
+                    logger.warn("Problem starting periodic Ping. {}", e.getMessage());
+                }
+            }, 25, 25, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void onWebSocketError(@NonNullByDefault({}) Throwable cause) {
+        Throwable localThrowable = (cause != null) ? cause
+                : new IllegalStateException("Null Exception passed to onWebSocketError");
+        logger.debug("Error during websocket communication: {}", localThrowable.getMessage());
+
+        Session localSession = session;
+        if (localSession != null) {
+            stopKeepAlive();
+            localSession.close(StatusCode.SERVER_ERROR, "Failure: " + localThrowable.getMessage());
+            session = null;
+        }
+
+        notifyListenersOnError(String.format("Error: %s", localThrowable.getMessage()));
+    }
+
+    @Override
+    public void onWebSocketBinary(@NonNullByDefault({}) byte[] payload, int offset, int len) {
+        throw new UnsupportedOperationException("Unimplemented method 'onWebSocketBinary'");
+    }
+
+    @Override
+    public void onWebSocketText(@NonNullByDefault({}) String message) {
+        if (message.contains("\"event\":\"statistics updated\"")) {
+            return;
+        }
+
+        BaseMessage baseEvent = null;
+        try {
+            baseEvent = gson.fromJson(message, BaseMessage.class);
+        } catch (Exception ex) {
+            logger.warn("Failed to parse incoming WebSocket message: {}", ex.getMessage());
+            logger.trace("RECV | {}", message);
+            notifyListenersOnError("Failed to parse message: " + ex.getMessage());
+            return;
+        }
+
+        if (baseEvent == null || baseEvent.type == null) {
+            logger.warn("Received event with unknown or null type.");
+            logger.trace("RECV | {}", message);
+            notifyListenersOnError("Received event with unknown or null type.");
+            return;
+        }
+
+        handleBaseEvent(baseEvent, message);
+
+        // Notify listeners
+        try {
+            for (ZwaveEventListener listener : listeners) {
+                listener.onEvent(baseEvent);
+            }
+        } catch (Exception e) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Error invoking event listener on websockettext: {}", e.toString());
+            } else {
+                logger.warn("Error invoking event listener on websockettext", e);
+            }
+        }
+
+        // Special handling for VersionMessage
+        if (baseEvent instanceof VersionMessage) {
+            // the binding is starting up, perform schema version handshake
+            // also start listening to events
+            sendCommand(new ServerInitializeCommand());
+            // sendCommand(new StatisticsCommand(false));
+            sendCommand(new ServerListeningCommand());
+        }
+    }
+
+    private void handleBaseEvent(BaseMessage baseEvent, String message) {
```

Comment:
This method is just logging information.
Maybe it could have a more appropriate name ?

Reply:
Yes, done